engine/access/ingestion/engine.go

```diff
@@ -424,6 +426,15 @@ func (e *Engine) handleExecutionReceipt(originID flow.Identifier, r *flow.Execut
 		return fmt.Errorf("failed to store execution receipt: %w", err)
 	}
 
+	block, err := e.blocks.ByID(r.ExecutionResult.BlockID)
+	if err != nil {
+		return fmt.Errorf("failed to lookup block id while handling receipt: %w", err)
+	}
+
+	if block.Header.Height > e.maxReceiptHeight {
+		e.transactionMetrics.UpdateExecutionReceiptMaxHeight(block.Header.Height)
```

Comment:
I missed this before, but I think you want to also update `maxReceiptHeight` here with the new highest height. Same in the execution engine

Reply:
ahhhh I can't believe I forgot this.